Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,490 @@
+		/// <exception cref="System.ArgumentException">Calendar does not exist on device</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
+		{
+			return Task.Run<IList<CalendarEvent>>(() =>
+			{
+				var calendars = new List<CalendarEvent>();
+
+				CalendarManager calManager = null;
+				CalendarList calList = null;
+				CalendarRecord calRecord = null;
+
+				if (calList.Count == 0)
+					return calendars;
+
+				calList.MoveFirst();
+
+				try
+				{
+					do
+					{
+						calManager = new CalendarManager();
+						calList = calManager.Database.GetAll(Event.Uri, 0, 0);
+						calRecord = calList.GetCurrentRecord();
+						string summary = calRecord.Get<string>(Event.Summary);
+						int IsAllday = calRecord.Get<int>(Event.IsAllday);
+						CalendarTime sTime = Convert.ToBoolean(IsAllday)
+								? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
+								: calRecord.Get<CalendarTime>(Event.Start);
+						CalendarTime End = Convert.ToBoolean(IsAllday)
+										? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
+										: calRecord.Get<CalendarTime>(Event.End);
+						string location = calRecord.Get<string>(Event.Location);
+						string description = calRecord.Get<string>(Event.Description);
+						int Id = calRecord.Get<int>(Event.Id);
+						TimeSpan s = sTime.UtcTime - start;
+						TimeSpan e = end - End.UtcTime;
+						if (s.TotalSeconds >= 0 && e.TotalSeconds >= 0)
+						{
+							calendars.Add(
+								new CalendarEvent
+								{
+									Name = summary,
+									Start = sTime.UtcTime,
+									End = End.UtcTime,
+									Location = location,
+									Description = description,
+									AllDay = Convert.ToBoolean(IsAllday),
+									ExternalID = Id.ToString(),
+								}
+							);
+						}
+					} while (calList.MoveNext());
+				}
+				catch (Exception ex)
+				{
+					throw new Exception(ex.Message);
+				}
+				finally
+				{
+					calRecord?.Dispose();
+					calRecord = null;
+
+					calList?.Dispose();
+					calList = null;
+
+					calManager?.Dispose();
+					calManager = null;
+				}
+
+				return calendars;
+			});
+		}
+
+		/// <summary>
+		/// Gets a single calendar event by platform-specific ID.
+		/// </summary>
+		/// <param name="externalId">Platform-specific calendar event identifier</param>
+		/// <returns>The corresponding calendar event, or null if not found</returns>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<CalendarEvent> GetEventByIdAsync(string externalId)
```

Comment:
This is ignoring the `externalId` parameter and thus is presumably not actually returning the requested event.